extensions/stackdriver/common/utils.cc

```diff
@@ -63,6 +63,20 @@ void getMonitoredResource(const std::string &monitored_resource_type,
   }
 }
 
+void setSTSService(
+    ::envoy::config::core::v3::GrpcService_GoogleGrpc_CallCredentials_StsService
+        *sts_service,
+    const std::string &sts_port) {
+  if (!sts_service) {
+    return;
+  }
+  sts_service->set_token_exchange_service_uri("http://127.0.0.1:" + sts_port +
```

Comment:
nit: are we OK with hard-coding IPv4 localhost here? this won't work on IPv6 machines.
Is there a reason for allowing only port changes?

Reply:
@JimmyCYJ ^^^

Reply:
@JimmyCYJ mind shed some lights on this?

Reply:
Yes, the server side could listen on [IPv6 address](https://github.com/istio/istio/blob/25ac96709925ecceb9ab79b6a325d031e6100733/pilot/cmd/pilot-agent/main.go#L512), is there a way to detect whether the Proxy is using IPv4 or IPv6 like Pilot-agent?

Reply:
If gRPC supports name lookups, then hopefully `localhost` could be used. Either way, we can't hardcode 127.0.0.1 I think.

Reply:
Switch to localhost. Right now at master and 1.5, sts server is running as within pilot-agent, and the only exposed option is port number. So configurability of other parts is not quite necessary.